THIRD-PARTY

```diff
@@ -8211,7 +8211,7 @@ The Apache Software Foundation (http://www.apache.org/).
 
 --------------------------------------------------------------------------------
 
-104. Group: org.apache.logging.log4j  Name: log4j-api  Version: 2.11.1
+104. Group: org.apache.logging.log4j  Name: log4j-api  Version: 2.15.0
```

Comment:
Can you explain why `log4j-api` version here is 2.15.0 whereas down below on line 8431 it's 2.16.0?

Reply:
This dependency file is auto generated based on the dependencies we use in the Data Prepper.

Reply:
I know it's auto generated, I'm thinking if there's some place we're using 2.15.0 in `log4j-api` dependency and if we should upgrade that to 2.16.0

Reply:
Other dependencies may be pulling it in. The authoritative way to determine the actual dependencies at runtime would be to run:

```
./gradlew -p data-prepper-core dependencies
```

Reply:
There may be a future opportunity to pare down some duplicate dependencies in the automation. But, I don't think having both versions in the THIRD-PARTY file is a problem right now.

Reply:
@dlvenable The actual dependencies for `log4j-api` at runtime show all versions as 2.16.0, so I believe we're good and unsure where that 2.15.0 is popping up from.
```
./gradlew -p data-prepper-core dependencies | grep log4j
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
\--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0 (n)
+--- org.apache.logging.log4j:log4j-core (n)
\--- org.apache.logging.log4j:log4j-slf4j-impl (n)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    |    |    +--- org.apache.logging.log4j:log4j-api:2.11.1 -> 2.16.0
|    |    |    |    +--- org.apache.logging.log4j:log4j-core:2.11.1 -> 2.16.0
|    |    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.16.0
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0 (*)
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    +--- org.apache.logging.log4j:log4j-api:2.16.0
|    \--- org.apache.logging.log4j:log4j-core:2.16.0 (*)
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    \--- org.apache.logging.log4j:log4j-api:2.16.0
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
\--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    |    |    +--- org.apache.logging.log4j:log4j-api:2.11.1 -> 2.16.0
|    |    |    |    +--- org.apache.logging.log4j:log4j-core:2.11.1 -> 2.16.0
|    |    |    |    |    \--- org.apache.logging.log4j:log4j-api:2.16.0
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    |    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-bom:2.16.0
|    +--- org.apache.logging.log4j:log4j-core:2.16.0 (c)
|    +--- org.apache.logging.log4j:log4j-slf4j-impl:2.16.0 (c)
|    \--- org.apache.logging.log4j:log4j-api:2.16.0 (c)
+--- org.apache.logging.log4j:log4j-core -> 2.16.0 (*)
+--- org.apache.logging.log4j:log4j-slf4j-impl -> 2.16.0
|    +--- org.apache.logging.log4j:log4j-api:2.16.0
|    \--- org.apache.logging.log4j:log4j-core:2.16.0 (*)
+--- org.apache.logging.log4j:log4j-core:2.15.0 -> 2.16.0 (c)
+--- org.apache.logging.log4j:log4j-api:2.15.0 -> 2.16.0 (c)

```